languages/go/exercises/concept/README.md

```diff
@@ -8,7 +8,7 @@ These are the concept exercises that have currently been implemented, as well as
 
 | exercise                                                            | concepts                                                       | prerequisites                                                  |
 | ------------------------------------------------------------------- | -------------------------------------------------------------- | -------------------------------------------------------------- |
-| -                                                                   | -                                                              | -                                                              |
+| strings                                                             | `basic-strings`, `conditionals`                                | -                                                              |
```

Comment:
Wondering if I should add `conditionals` as a concept here or not. This exercise doesn't really teach it, but it is useful if doing it right. In this case I guess about 30% will use an `if`, the others won't. 

We could force students to have to use it by adding another test for the `Message` function with input "[ERROR]" (no message). That would `panic` if solved with `strings.Split` and without a check.

If students solve it by looking for ":" and it is not found, then they would also need an `if` statement if we add above test case.

What do you think?

Reply:
@ErikSchierboom What do you think?

Reply:
Note: if we force students to use it here, we don't need to introduce it again later. If we don't we have to deal with it again.

Reply:
Hmmm, tough call. You say that about 30% will use an `if`. What is the canonical solution you'd want to guide students to? If that solution uses an `if`, an analyzer could help point this out.

Reply:
I'd say the canonical version has a check if it doesn't try to access an index that does not exist. It would `panic` otherwise which in Go is a bug to be fixed.

Nice idea with the `analyzer` instead of adding a test.

Reply:
this is what I'd consider canonical:

```go
func Message(line string) string {
	parts := strings.Split(line, ":")
	if len(parts) < 2 {
		return ""
	}

	return strings.TrimSpace(parts[1])
}
```

Reply:
or this:

```go
func Message(line string) string {
	pos := strings.Index(line, ":")
	if pos == -1 {
		return ""
	}

	return strings.TrimSpace(line[pos+1:])
}
```

Reply:
To me it would then make sense to include it as a concept.

@iHiD Any thoughts?

Reply:
Yeah, definitely. You're hoping for an `if`, so I think mention that. You're also assigning a variable, using a colon :o ;) So maybe saying something like this:

> This exercise will introduce you to [assignment](...), [conditionals](...), [String functions](...) and [explicit returns](..). 

That might be too "brute force" but I think linking out to relevant docs in early exercises is good to familiarise students with where to look for things.

Reply:
Didn't see all those concepts, but now that you mention it 😄 

It might need some exercises before this one to have fewer new concepts in this one. e.g. introducing `functions`, `return statements` and `assignments` earlier.